content/docs/examples/multicluster/split-horizon-eds/index.md

```diff
@@ -0,0 +1,377 @@
+
+    {{< text yaml >}}
+    apiVersion: v1
+    kind: Service
+    metadata:
+      name: helloworld
+      labels:
+        app: helloworld
+    spec:
+      ports:
+      - port: 5000
+        name: http
+      selector:
+        app: helloworld
+    ---
+    apiVersion: extensions/v1beta1
+    kind: Deployment
+    metadata:
+      name: helloworld-v2
+    spec:
+      replicas: 1
+      template:
+        metadata:
+          labels:
+            app: helloworld
+            version: v2
+        spec:
+          containers:
+          - name: helloworld
+            image: istio/examples-helloworld-v2
+            imagePullPolicy: IfNotPresent
+            ports:
+            - containerPort: 5000
+    {{< /text >}}
+
+1. Deploy the file:
+
+    {{< text bash >}}
+    $ kubectl create --context=$CTX_REMOTE -f helloworld-v2.yaml -n sample
+    {{< /text >}}
+
+### Deploy helloworld v1 in local
+
+1. Create a `sample` namespace with a sidecar auto-injection label:
+
+    {{< text bash >}}
+    $ kubectl create --context=$CTX_LOCAL ns sample
+    $ kubectl label --context=$CTX_LOCAL namespace sample istio-injection=enabled
+    {{< /text >}}
+
+1. Create a file `helloworld-v1.yaml` with the following content:
+
+    {{< text yaml >}}
+    apiVersion: v1
+    kind: Service
+    metadata:
+      name: helloworld
+      labels:
+        app: helloworld
+    spec:
+      ports:
+      - port: 5000
+        name: http
+      selector:
+        app: helloworld
+    ---
+    apiVersion: extensions/v1beta1
+    kind: Deployment
+    metadata:
+      name: helloworld-v1
+    spec:
+      replicas: 1
+      template:
+        metadata:
+          labels:
+            app: helloworld
+            version: v1
+        spec:
+          containers:
+          - name: helloworld
+            image: istio/examples-helloworld-v1
+            imagePullPolicy: IfNotPresent
+            ports:
+            - containerPort: 5000
+    {{< /text >}}
+
+1. Create a file `helloworld-gateway.yaml` with the following content:
+
+    {{< text yaml >}}
+    apiVersion: networking.istio.io/v1alpha3
+    kind: Gateway
+    metadata:
+      name: helloworld-gateway
+      namespace: sample
+    spec:
+      selector:
+        istio: ingressgateway
+      servers:
+      - port:
+          number: 443
+          name: tls
+          protocol: TLS
+        tls:
+          mode: AUTO_PASSTHROUGH
+        hosts:
+        - "*"
+    {{< /text >}}
+
+    Although deployed locally this Gateway instance will also affect the remote cluster by configuring it to passthrough incoming traffic to the relevant remote service (SNI-based) but keeping the mTLS all the way from the source to destination sidecars.
+
+1. Deploy the files:
+
+    {{< text bash >}}
+    $ kubectl create --context=$CTX_LOCAL -f helloworld-v1.yaml -n sample
+    $ kubectl create --context=$CTX_LOCAL -f helloworld-gateway.yaml -n sample
+    {{< /text >}}
```

Comment:
Any virtual service and dr need to be deployed?  I don't see how you bind the helloworld gateway to the helloworld service.  Or is that not needed with SNI based routing?

Reply:
Nope. Using the new AUTO_PASSTHROUGH tls mode in the `Gateway` object. It is also being used for the gateways-based multicluster.

Reply:
Ok I have one more q, how does AUTO_PASSTHROUGH tls mode in the gateway object knows to pass through to the helloworld service when you have multiple sample services deployed in the sample namespace?

Reply:
SNI-based